compat/wserrno.cc

```diff
@@ -19,27 +19,22 @@ SetErrnoFromWsaError()
     // POSIX codes which socket API users may care about
     static const auto *CodeMap = new std::unordered_map<int, int> {
 
-        // values checked for by accept(2) callers
         { WSAECONNABORTED, ECONNABORTED },
 
-        // values checked for by connect(2) callers
         { WSAEINPROGRESS, EINPROGRESS },
         { WSAEAFNOSUPPORT, EAFNOSUPPORT },
         { WSAEINVAL, EINVAL },
         { WSAEISCONN, EISCONN },
 
-        // values checked by ignoreErrno()
         { WSAEWOULDBLOCK, EWOULDBLOCK },
-        // WSAEAGAIN not defined
+        // WSAEAGAIN has no equivalent
```

Comment:
Has it not? We do use EAGAIN elsewhere:

```suggestion
        { WSAEAGAIN, EAGAIN },
```

Reply:
@kinkie, please restore those "not defined" comments. The "has no equivalent" means something different and is wrong for this mapping. We only need to map the __existing__ `WSAE...` codes.

Reply:
@kinkie, please do not restore those "not defined" comments "as is" because that will resurrect my earlier change request. To move forward, if you want to add comments that document what codes are currently not listed on Microsoft web site, then do something like this (after replacing XXX placeholder with an explanation):

```C++
// Maps Windows Error Codes to POSIX codes that socket API users may care about.
// All "is not listed" comments below mark Windows error codes that are
// not listed as "Windows Sockets Error Codes" in Microsoft documentation.
// We do not map those codes because XXX.
static const auto CodeMap = new std::unordered_map<int, int> {
...
        // WSAEAGAIN is not listed
```

Personally, I still do not know why we do not want to map those unlisted codes, so I cannot address that XXX myself, but it should be addressed if we restore some sort of "unlisted" comment instead of mapping.

Reply:
> > Alex: Personally, I still do not know why we do not want to map those unlisted codes, so I cannot address that XXX
> 
> > Amos: Because compiler output: `` error: undefined symbol `WSAEAGAIN` ``
> 
> To clarify: My comment above does _not_ apply to WSAfoo codes that do not exist.

The "not exist" of some WSAEfoo codes is the fundamental basis of this discussion.
... Your **design change** insistence on documentation stating that non-existence into documentation of some other meaning(s). see my below comment.

> To avoid red herring chases, non-existing code names should never be mentioned at all! My comment only applies to _existing_ WSAfoo codes that are not listed as "Windows Sockets Error Codes" in Microsoft documentation. If there are no such WSAfoo codes (that are relevant here), then my "is not listed" compromise is not needed at all, which is a good thing!
> 

"If"  ... @rousskov, you are imposing design decisions and choices on us without understanding the API and logic/code being impacted by those decisions. This has, as usual, resulted in a series of long discussions and argument about minor details. In short; a huge waste of all our time.



> For cases where no WSAfoo code exists, the "document why we do not map to Efoo" problem needs a different solution: If POSIX code has no WSAfoo equivalent, then we should document that fact using POSIX code name. We should not invent WSAfoo names.

No, it does not need any fancy criteria. Just state the particular code is not defined. Period. No need for arguments or new policy rules.

Reply:
> I've reworded the EAGAIN comment

Thank you. I am resolving this change request because commit e5541d3c167651d6a32dd7841acd6af3ae8c937f addresses it. The new comment has problems of its own, but they are different from the ones discussed here, so I will open a new change request to address them.

> I fail to see what the problem is.

AFAICT, the problem was identified in the change request: The proposed "WSAEAGAIN has no equivalent" comment was unclear and/or misleading because (made up, but that was not clear either) WSAEAGAIN does have an equivalent called EAGAIN. Thank you for fixing that problem. In the future, if you do not see what the alleged problem is, please state that and ask for clarifications!